instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/handlers/enqueue.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActiveJob
+      module Handlers
+        # Handles `enqueue.active_job` and `enqueue_at.active_job` to generate egress spans
+        class Enqueue < Default
+          def initialize(...)
+            super
+            @span_name_formatter = if @config[:span_naming] == :job_class
+                                     ->(job) { "#{job.class.name} publish" }
+                                   else
+                                     ->(job) { "#{job.queue_name} publish" }
+                                   end
```

Comment:
This is causing a regression in our codebase because the handler no longer honours dynamic configuration.

i.e. Changing `OpenTelemetry::Instrumentation::ActiveJob::Instrumentation.instance` after this is registered no longer affects the naming.

Arguably, we probably shouldn't rely on dynamic configs, but I wanted to point it out regardless

Same applies to `Perform`

Reply:
Would you be able to provide us with a PR to fix this issue?

Reply:
https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1079